lib/rspec/mocks/example_methods.rb

```diff
@@ -282,6 +282,17 @@ def have_received(method_name, &block)
         Matchers::HaveReceived.new(method_name, &block)
       end
 
+      # Turns off the verifying of partial doubles for the duration of the
+      # block, this is useful in situations where methods are defined at run
+      # time and you wish to define stubs for them but not turn off partial
+      # doubles for the entire run suite. (e.g. view specs in rspec-rails).
+      def without_verifying_partial_doubles
+        Mocks.configuration.temporarily_suppress_partial_double_verification = true
+        yield
+      ensure
+        Mocks.configuration.temporarily_suppress_partial_double_verification = false
+      end
```

Comment:
IMO, this should set the config flag back to whatever its value was before the method was called, rather than assuming it was set to `false` (which may or may not be true).  Consider a case like:

``` ruby
module MyHelpers
  def do_stuff
    without_verifyiong_partil_doubles do
      # ...
    end
  end
end

RSpec.describe MyGroup do
  include MyHelpers

  it "works" do
    without_verifying_partial_doubles do
      do_something
      do_stuff
      do_something_else
    end
  end
end
```

As you've implemented this, partial double verification will _not_ be disabled while `do_something_else` runs, even though its in a `without_verifying_partial_doubles` block.  I'd argue this is a bug.

The solution is to change this method to stash the existing value of the config flag when the method starts, then set it to true, then set it back to the original value in the `ensure` block.